src/shared/generator.c

```diff
@@ -21,41 +21,58 @@
 #include "specifier.h"
 #include "string-util.h"
 #include "time-util.h"
+#include "tmpfile-util.h"
 #include "unit-name.h"
 
-int generator_open_unit_file(
+int generator_open_unit_file_full(
                 const char *dir,
                 const char *source,
                 const char *fn,
-                FILE **ret) {
+                FILE **ret_file,
+                char **ret_temp_path) {
 
         _cleanup_free_ char *p = NULL;
         FILE *f;
         int r;
 
         assert(dir);
-        assert(fn);
-        assert(ret);
+        assert(ret_file);
 
-        p = path_join(dir, fn);
-        if (!p)
-                return log_oom();
+        /* If <ret_temp_path> is specified, it creates a temporary unit file and also returns its
+         * temporary path. */
 
-        r = fopen_unlocked(p, "wxe", &f);
-        if (r < 0) {
-                if (source && r == -EEXIST)
-                        return log_error_errno(r,
-                                               "Failed to create unit file '%s', as it already exists. Duplicate entry in '%s'?",
-                                               p, source);
+        if (ret_temp_path) {
+                r = fopen_temporary(dir, &f, &p);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to create temporary unit file in '%s': %m", dir);
+
+                (void) fchmod(fileno(f), 0644);
 
-                return log_error_errno(r, "Failed to create unit file '%s': %m", p);
+                *ret_temp_path = TAKE_PTR(p);
+        }
+        else {
```

Comment:
```suggestion
        } else {
```

as per our coding style

Reply:
Fixed, thanks!